File: packages/ruviz-web/examples/kde.ts
import { createPlot } from "../src/index.js";
import { sampleDistribution } from "./data.js";
import type { PlotExampleDefinition } from "./types.js";

export const example: PlotExampleDefinition = {
  slug: "kde",
  title: "Kernel density estimate",
  summary: "A smoothed density curve for a numeric sample.",
  category: "statistical",
  createPlot() {
    return createPlot().sizePx(640, 320).title("Kernel Density Estimate").xlabel("value").kde(sampleDistribution());
  },
};

File: bindings/python/examples/kde.py
from __future__ import annotations

from _shared import ExampleMeta, base_plot, sample_distribution, save_example

META = ExampleMeta(
    slug="kde",
    title="Kernel density estimate",
    summary="A smoothed density curve for a numeric sample with an explicit bandwidth.",
    section="Statistical plots",
)


def build_plot():
    return (
        base_plot("Kernel Density Estimate")
        .xlabel("value")
        .kde(sample_distribution(), bandwidth=0.35, color="#7c3aed", width=2.0)
    )


if __name__ == "__main__":
    save_example(META, build_plot())
